### DISCUSSION TITLE
J1708🎉

### DISCUSSION THREAD

**Discussion by BenGardiner** [dfieschko/RP1210#85] [Show and tell]
Thanks for this library. I'm able to use J1708 on an RP1210 adapter by monkeypatching clienID

```python
api = RP1210.RP1210API(self.dll_name)
config = RP1210.RP1210Config(self.dll_name)
device_id = config.getProtocol("J1708").getDevices()[0]
client_id = api.ClientConnect(device_id, b"J1708:Baud=9600")

client = RP1210.RP1210Client()
client.setVendor(self.dll_name)
client.clientID = client_id
```

then `client` is ready to go. 

`rx()` return the expected 4 bytes of timestamps which need to be discarded and `tx()` expects a byte prepended which is discarded before sending the rest.

**Comment by dfieschko** [comment]
Very nice! I'd like to note that RP1210Client is a child of RP1210VendorList, which stores instances of all available RP1210Config objects in a list (which is populated via the API names in RP121032.ini in C:\Windows). RP1210Client allows you to select which vendor and which device you want to use (accessed via `getCurrentVendor()` and `getCurrentDevice()`), and is built for use with dropdown menus.

RP1210Client initializes its own instances of RP1210API and RP1210Config, which can be accessed via `client.getAPI()` and `client.getCurrentVendor()`. Also, RP1210Client has its own `connect` function that calls `RP1210_ClientConnect` and automatically stores clientID.

```python
client = RP1210.RP1210Client()
 # set vendor and device (intended for use with a dropdown menu):
client.setVendor(self.dll_name)
device_id = client.getCurrentVendor().getProtocol("J1708").getDevices()[0]
client.setDevice(device_id)
# connect to device that was selected in previous step:
client.connect(b"J1708:Baud=9600") # this returns clientID, but will also automatically store in client.clientID
```

This functionality would be more clear if I populated the `RP1210Client` section of the wiki. I should get around to doing that.

This discussion made me realize that it'd be convenient to have the current RP1210API and RP1210Config list accessible as properties; I'll add an issue for that. It would also be nice to break out current device as a property.

I should probably also add support for selecting protocol and baud with RP1210Client, which would eliminate the need for the user of the package to generate their own protocol string.

Thanks for using the package - your comment has been a big help to me just by showing me how you used it.

**Reply by dfieschko** [reply]
This may be the impetus for me adding J1708.py along with J1939.

**Reply by dfieschko** [reply]
See #86 #87 #88

**Reply by BenGardiner** [reply]
hi @dfieschko we're integrating RP120 support via this library into py-hv-networks and some other tools e.g. https://github.com/TruckHacking/py-hv-networks/pull/7

**Reply by dfieschko** [reply]
> hi @dfieschko we're integrating RP120 support via this library into py-hv-networks and some other tools e.g. [TruckHacking/py-hv-networks#7](https://github.com/TruckHacking/py-hv-networks/pull/7)

Very cool! Let me know if you need any help, and please give me any suggestions you have!
